### DISCUSSION TITLE
`class Opt(Enum)` as default value is not handled correctly

### DISCUSSION THREAD

**Discussion by trim21** [fastapi/typer#775] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from enum import Enum

import typer


class NeuralNetwork(Enum):
    simple = "simple"
    conv = "conv"
    lstm = "lstm"


def main(network: NeuralNetwork = NeuralNetwork.simple):
    print(f"Training neural network of type: {network.value}")


if __name__ == "__main__":
    typer.run(main)
```

### Description

this result in to a error:

```text
┌─ Error ─────────────────────────────────────────────────────────────────────┐
│ Invalid value for '--network': <NeuralNetwork.simple: 'simple'> is not one  │
│ of 'simple', 'conv', 'lstm'.                                                │
└─────────────────────────────────────────────────────────────────────────────┘
```

### Operating System

Linux

### Typer Version

0.11.1

### Python Version

Python 3.11.6

**Comment by AlessandroPomponio** [comment]
Hi @trim21 !
Try adding `.value` to your default and it should work:

```python
def main(network: NeuralNetwork = NeuralNetwork.simple.value):
```

Note: I think this is required because the `convert` method that gets invoked expects a string to convert into your Enum type, so the default will need to be the related string value but in the code you'll get the Enum ^^

**Reply by trim21** [reply]
I think this is a issue becasue both `class NeuralNetwork(enum.IntEnum)`, `class NeuralNetwork(int, enum.Enum)` `class NeuralNetwork(enum.StrEnum)` and `class NeuralNetwork(str, enum.Enum)` work with `network: NeuralNetwork = NeuralNetwork.simple`

**Reply by zack-mayoh** [reply]
This creates a new type issue. It results in this error from mypy:
```
Incompatible default for argument "network" (default has type "str", argument has type "NeuralNetwork")
```
